fuzz/src/lib.rs

```diff
@@ -64,90 +79,77 @@ pub trait DbSimulator {
 		);
 		for action in &actions {
 			// We apply the action on both the database and the model
+			log::debug!("Applying on the database: {:?}", action);
 			match action {
 				Action::Transaction(operations) => {
-					if let Err(e) =
-						db.commit_changes(operations.iter().map(|o| (0, Self::map_operation(o))))
-					{
-						drop(db);
-						let (d, m) = Self::fail_or_restart_after_db_error(
-							e,
-							&options,
-							&actions,
-							config.btree_index,
-						);
-						model = m;
-						db = d;
-					} else {
-						for o in operations {
-							Self::apply_operation_on_model(o, &mut model);
-						}
-						retry_operation(|| {
-							Self::check_db_and_model_are_equals(&db, &model, config.btree_index)
-						})
+					Self::apply_operations_on_model(operations, &mut model);
+					db.commit_changes(operations.iter().map(|o| (0, Self::map_operation(o))))
 						.unwrap();
+				},
+				Action::ProcessReindex =>
+					db = Self::try_or_restart(|db| db.process_reindex(), db, &mut model, &options),
+				Action::ProcessCommits => {
+					Self::write_first_layer_to_disk(&mut model);
+					db = Self::try_or_restart(|db| db.process_commits(), db, &mut model, &options)
+				},
+				Action::FlushAndEnactLogs => {
+					for _ in 0..2 {
```

Comment:
Could use a comment

Reply:
Done

Reply:
Side note: it might  be relevant to refactor a bit the log system to allow independant enact and flush

Reply:
@Tpt I feel like you have force push over your changes (or just github ui bugging).

Reply:
It's the correct place I think. I have just force pushed a slight code style fix on my last commit.

Reply:
Indeed, lot of my review comment were  done before https://github.com/paritytech/parity-db/pull/143/commits/832c889dd20ab1d89b633223470c3d2a81a0e8aa (not sure why I don't have a refresh view on github).